File: ABC/abc021/abc021_d 1.py
#99点解法
n = int(input())
k = int(input())

ans = [0] * (n+1) #ans[k][i] := 1 ≤ a_1 ≤ ... ≤ a_k ≤ i を満たす整数の組aの個数(累積和) → 二次元を一次元にしたもの
ans[1] = 1
for _ in range(k+1):
    for i in range(n):
        ans[i+1] = (ans[i+1] + ans[i]) % 1000000007

print(ans[n])

File: ABC/abc021/abc021_d 1.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    // 99点解法
    int n, k;
    cin >> n >> k;

    vector<long long> ans(n+1, 0); // ans[k][i] := 1 ≤ a_1 ≤ ... ≤ a_k ≤ i を満たす整数の組aの個数(累積和) → 二次元を一次元にしたもの
    ans[1] = 1;
    for (int i = 0; i <= k; i++) {
        for (int j = 0; j < n; j++) {
            ans[j+1] = (ans[j+1] + ans[j]) % 1000000007;
        }
    }

    cout << ans[n] << endl;
}